units/unit.cpp

```diff
@@ -14,19 +15,33 @@
 
 namespace scipp::units {
 
+namespace {
+std::string map_unit_string(const std::string &unit) {
+  // custom dimensionless name
+  return unit == "dimensionless" ? ""
+                                 // use Gregorian years by default
+                                 : unit == "y" || unit == "year" ? "a_g" : unit;
+}
+} // namespace
+
 Unit::Unit(const std::string &unit)
-    : Unit(llnl::units::unit_from_string(unit == "dimensionless" ? "" : unit)) {
+    : Unit(llnl::units::unit_from_string(map_unit_string(unit))) {
   if (!is_valid(m_unit))
     throw except::UnitError("Failed to convert string `" + unit +
                             "` to valid unit.");
 }
 
 std::string Unit::name() const {
+  if (*this == Unit{"month"}) {
+    return "M";
+  }
   auto repr = to_string(m_unit);
   repr = std::regex_replace(repr, std::regex("^u"), "µ");
   repr = std::regex_replace(repr, std::regex("item"), "count");
   repr = std::regex_replace(repr, std::regex("count(?!s)"), "counts");
-  return repr == "" ? "dimensionless" : repr;
+  repr = std::regex_replace(repr, std::regex("day"), "D");
+  repr = std::regex_replace(repr, std::regex("a_g"), "Y");
```

Comment:
`map_unit_string` only handles `Y`, does the roundtrip work?

Reply:
No, it doesn't, good catch!
Is it bad that round trip does not work for 'day' or 'a_g'? I guess not since the older replacements also broke it.

Reply:
I suppose it depends on which roundtrip we are talking about:
- `Unit` -> string -> `Unit` or
- string -> `Unit` -> string

Of course for the latter it will convert to canonical representation, but the former should always work?